homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -70,21 +67,13 @@ def __init__(
         surepy_entity: SurepyEntity = self._spc.states[self._id]
 
         # cover special case where a device has no name set
-        if surepy_entity.name:
-            name = surepy_entity.name
-        else:
-            name = f"Unnamed {surepy_entity.type.name.capitalize()}"
-
-        self._name = f"{surepy_entity.type.name.capitalize()} {name.capitalize()}"
+        type_name = surepy_entity.type.name.replace("_", " ").title()
+        name = surepy_entity.name if surepy_entity.name else f"Unnamed {type_name}"
 
+        self._attr_name = f"{type_name} {name}"
```

Comment:
This will change the name of the sensors? Intendeed?

Reply:
You talk about the `.replace("_", " ").title()` above the line you marked?

Reply:
I wonder if the name will change? Is that intended?

Reply:
If the type is a Cat Flap or Pet Flap, the type.name will have a _ instead of a whitespace.
The _ will be replaced by a whitespace then. So, how do we classify this? breaking change? fix? And what would break in your eyes? I don't really get it 🤔

Reply:
If the name change, the entity id might change, and then automation might break for the users.
It is a bit difficult to understand how the name will change here.